custom_metrics/Images.js

```diff
@@ -2,19 +2,26 @@ var wptImages = function(win) {
   var images = [];
   if (win) {
     var doc = win.document;
-    var elements = doc.getElementsByTagName('*');
-    var re = /url\((http.*)\)/ig;
+    var elements = doc.getElementsByTagName("*");
     for (var i = 0; i < elements.length; i++) {
       var el = elements[i];
-      if (el.tagName == 'IMG') {
+      if (el.tagName == "IMG") {
         var url = el.currentSrc || el.src;
 
         // Only include HTTP(S) URLs i.e. skip dataURIs
-        if(url.indexOf("http") === 0) {
-          images.push({'url': url, 'width': el.width, 'height': el.height, 'naturalWidth': el.naturalWidth, 'naturalHeight': el.naturalHeight});
+        if (url.indexOf("http") === 0) {
+          images.push({
+            url: url,
+            width: el.width,
+            height: el.height,
+            naturalWidth: el.naturalWidth,
+            naturalHeight: el.naturalHeight,
+            loading: el.getAttribute("loading"),
+            "in-viewport": el.getBoundingClientRect().top < window.innerHeight,

```

Comment:
@kevinfarrugia a suggestion by Colin Bendell in the webperf Slack was to also consider the `left` property to handle carousels that shift images offscreen. Would you be able to open a new PR to update the script, taking that into consideration? Another idea is to use IntersectionObserver but that might be overkill.

Reply:
I'd also suggest looking at all corners of the image to see if they are within the viewport and not negatively offset out of the top or left

Reply:
Yes sure, could be done. Will try have a PR by Monday although I think I will immediately opt for the IntersectionObserver as this would probably be a more robust solution.